src/instructlab/model/train.py

```diff
@@ -520,20 +585,328 @@ def get_files(directory: str, pattern: str) -> list[str]:
+    # Third Party
+    from instructlab.eval.mmlu import MMLU_TASKS, MMLUEvaluator
+    import torch
+
+    evaluator = MMLUEvaluator(ckpt_path, tasks=MMLU_TASKS, few_shots=5)
+
+    mmlu_results: tuple[float, dict] = evaluator.run()
+    ckpt_score: float = mmlu_results[0]
+
+    logging.debug("Phased Training -- MMLU eval phase -- Clearing PyTorch cache")
+    torch.cuda.empty_cache()
+
+    return ckpt_score
+
+
+def _mtbench(ctx, ckpt_path: str, eval_cache_path: str, judge_name: str) -> float:
+    # TODO: run all generations in serial and then do all judgements at once to save time loading/unloading prometheus.
+
+    # dynamic import- eval code not used outside of this path.
+    # Third Party
+    from instructlab.eval.mt_bench import MTBenchEvaluator
+    import torch
+
+    # First Party
+    from instructlab.model.evaluate import launch_server
+
+    # hard-override for local insecure vLLM serving
+    ctx.params["tls_client_cert"] = None
+    ctx.params["tls_client_key"] = None
+    ctx.params["tls_client_passwd"] = None
+    ctx.params["tls_insecure"] = True
+
+    evaluator = MTBenchEvaluator(
+        model_name=ckpt_path,
+        judge_model_name=judge_name,
+        output_dir=eval_cache_path,
+        merge_system_user_message=True,
+    )
+
+    server = None
+    model_serve_url = None
+    try:
+        logger.debug("Starting model server for mt-bench answer generation")
+        server, model_serve_url = launch_server(
+            ctx=ctx,
+            model=ckpt_path,
+            model_name=ckpt_path,
+            gpus=torch.cuda.device_count(),
+            max_workers=multiprocessing.cpu_count(),
+            enable_serving_output=False,
+            backend="vllm",
+        )
+        logger.debug("Generating mt-bench answers")
+        evaluator.gen_answers(model_serve_url)
+    finally:
+        if server is not None:
+            server.shutdown()
+
+    try:
+        logger.debug("Starting model server for mt-bench answer judgement")
+        server, model_serve_url = launch_server(
+            ctx=ctx,
+            model=judge_name,
+            model_name=judge_name,
+            gpus=torch.cuda.device_count(),
+            max_workers=multiprocessing.cpu_count(),
+            backend="vllm",
+            enable_serving_output=False,
+        )
+        logger.debug("Judging mt-bench answers")
+        mt_bench_results: tuple = evaluator.judge_answers(model_serve_url)
+        ckpt_score: float = mt_bench_results[0]
+    finally:
+        if server is not None:
+            server.shutdown()
+
+    return ckpt_score
+
+
+def _evaluate_dir_of_checkpoints(
+    checkpoints_dir_path: str, eval_func: typing.Callable[..., float]
+) -> tuple[float, str]:
+    """Run eval_func on all model checkpoints in a directory, returning the best performing."""
+
+    # TODO: parallelize MMLU over available GPUs
+
+    results: list[typing.Tuple[float, str]] = []
+    directory = Path(checkpoints_dir_path)
```

Comment:
Why not have `checkpoints_dir_path` as a pathlib.Path in the first place?

Reply:
Right now, it's because there's more agreement to use str for paths than Path lib objects in the other modules. We could move this str->Path conversion up in the script (into _run_phased_training) but we're only using the iterdir feature here. It's sort of an agreement thing- Training uses path-strings, eval uses a lot of path-strings, etc, so I'm trying to stay in the same style.